### PR TITLE
WIP Refactor filtering moving from CLI to server

### PR DESCRIPTION
**Pull request by jwhonce**
* Filter on server side to reduce amount of data pushed over Varlink

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: libpod/runtime_ctr.go [hunk 1]
```diff
@@ -436,7 +437,7 @@ func (r *Runtime) LookupContainer(idOrName string) (*Container, error) {
 // Filters can be provided which will determine what containers are included in
 // the output. Multiple filters are handled by ANDing their output, so only
 // containers matching all filters are returned
-func (r *Runtime) GetContainers(filters ...ContainerFilter) ([]*Container, error) {
+func (r *Runtime) GetContainers(filters ...filterable.Filter) ([]*Container, error) {
```
Comment (mheon) [inline libpod/runtime_ctr.go]: I think we can keep the definition for ContainerFilter in libpod - anything that matches the signature will be treated by Go as a ContainerFilter, even if it's defined in another package

Comment (mheon) [reply libpod/runtime_ctr.go]: Oh, I see you redefined it. Hmmm.

Can we have filterable.Filter produce a ContainerFilter instead? I kind of like arbitrary-function interface we have going on here, and I'd like to keep it if we're retaining filtering.

**Comment by rhatdan** [comment]
@jwhonce Are you still working on this?

**Comment by rhatdan** [comment]
@jwhonce What is the skoop on this one?
@baude Do you know what is going on?

**Comment by jwhonce** [comment]
We went a different direction.

**Pull request closed without merging**